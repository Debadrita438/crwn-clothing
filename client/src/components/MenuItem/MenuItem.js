import { withRouter } from 'react-router';

import { 
    BackgroundImageContainer, 
    ContentContainer, 
    MenuItemContainer, 
    SubtitleStyles, 
    TitleStyles 
} from './MenuItem.styles';

export const MenuItem = ({ title, imageUrl, size, history, linkUrl, match }) => (
    <MenuItemContainer size={ size }
        onClick={ () => history.push(`${match.url}${linkUrl}`) }
    >
        <BackgroundImageContainer imageUrl={ imageUrl } className='background-image' />
        <ContentContainer>
            <TitleStyles>{ title.toUpperCase() }</TitleStyles>
            <SubtitleStyles>SHOP NOW</SubtitleStyles>
        </ContentContainer>
    </MenuItemContainer>
);
 
export default withRouter(MenuItem);