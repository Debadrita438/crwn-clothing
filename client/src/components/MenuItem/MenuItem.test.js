import { shallow } from 'enzyme';

import { MenuItem } from './MenuItem';

describe('MenuItem Component', () => {
    let wrapper;
    let mockHistory;
    let mockMatch;
    let linkUrl = '/womens';
    let imageUrl = 'www.testImage.com';
    let size = 'large';

    beforeEach(() => {
        mockHistory = {
            push: jest.fn()
        };

        mockMatch = {
            url: '/shop'
        };

        const mockProps = {
            title: 'Womens',
            imageUrl,
            size,
            history: mockHistory,
            linkUrl,
            match: mockMatch
        };

        wrapper = shallow(<MenuItem { ...mockProps } />);
    });

    it('should render MenuItem Component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call history.push with the right string when the MenuItemContainer is clicked', () => {
        wrapper.find('MenuItemContainer').simulate('click');
        expect(mockHistory.push).toHaveBeenCalledWith(`${ mockMatch.url }${ linkUrl }`);
    });

    it('should show the proper size', () => {
        expect(wrapper.find('MenuItemContainer').prop('size')).toBe(size);
    });

    it('should render the proper image', () => {
        expect(wrapper.find('BackgroundImageContainer').prop('imageUrl')).toBe(imageUrl);
    });


});