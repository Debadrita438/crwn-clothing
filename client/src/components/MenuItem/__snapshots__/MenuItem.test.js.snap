// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem Component should render MenuItem Component 1`] = `ShallowWrapper {}`;
